server/app/views/ViewUtils.java

```diff
@@ -301,4 +302,21 @@ public static ButtonTag makeToggleButton(
     text.ifPresent(button::withText);
     return button;
   }
+
+  public static DivTag makeUniversalBadge(QuestionType questionType) {
```

Comment:
suggestion(non-blocking):  This could easily be a more generic `makeBadge(String content)` so that we would have a more content-agnostic and reusable UI component.  I do see that there's already a `makeBadge` function.  That could be renamed to `makeProgramStatusBadge`.